src/pubsub/unsubscribe.js

```diff
@@ -57,5 +57,26 @@ module.exports = (createCommon, options) => {
         )
       })
     })
+
+    it('should subscribe 10 handlers and unsunscribe once with no reference to the handlers', (done) => {
+      const count = 10
+      const someTopic = getTopic()
+
+      timesSeries(count, (_, cb) => {
+        const handler = (msg) => {}
+        ipfs.pubsub.subscribe(someTopic, handler, (err) => cb(err))
+      }, (err) => {
+        expect(err).to.not.exist()
+        ipfs.pubsub.unsubscribe(someTopic)
+        setTimeout(() => {
+          // Assert unsubscribe worked
+          ipfs.pubsub.ls((err, topics) => {
+            expect(err).to.not.exist()
+            expect(topics).to.eql([])
+            done()
+          })
+        }, 500)
+      })
+    })
```

Comment:
Can we please use `async/await` or promises directly in this test? 500ms delay isn't guarateed to be enough time to wait.

Reply:
done